### PR TITLE
Marlin downstream PR

### PR DESCRIPTION
**Pull request by alexm-nm**

### CODE REVIEW

File: csrc/ops.h [hunk 1]
```diff
@@ -2,53 +2,41 @@
...
-  torch::Tensor& key_cache,
-  torch::Tensor& value_cache,
-  int num_kv_heads,
-  float scale,
-  torch::Tensor& block_tables,
-  torch::Tensor& context_lens,
-  int block_size,
-  int max_context_len,
-  const c10::optional<torch::Tensor>& alibi_slopes);
+void paged_attention_v1(torch::Tensor &out, torch::Tensor &query,
```
Comment (LucasWilkinson) [inline csrc/ops.h:5]: we should probably avoid reformatting this file, it'll cause headaches later on when syncing with main vLLM repo

File: vllm/model_executor/layers/parameters/sparsity.py [hunk 2]
```diff
@@ -1,29 +1,35 @@
 import torch
 
-from magic_wand import SparseTensor, SparseBitmaskStorageFormat
+from typing import Type
```
Comment (LucasWilkinson) [inline vllm/model_executor/layers/parameters/sparsity.py:3]: why is this file changing? this seems unrelated to Marlin

File: csrc/pybind.cpp [hunk 3]
```diff
@@ -51,11 +51,12 @@ PYBIND11_MODULE(TORCH_EXTENSION_NAME, m) {
 #ifndef USE_ROCM
   // Quantization ops
   ops.def("awq_gemm", &awq_gemm, "Quantized GEMM for AWQ");
+  ops.def("marlin_gemm", &marlin_gemm, "Marlin Optimized Quantized GEMM for GPTQ");
 #endif
   ops.def("gptq_gemm", &gptq_gemm, "Quantized GEMM for GPTQ");
   ops.def("gptq_shuffle", &gptq_shuffle, "Post processing for GPTQ");
   ops.def("squeezellm_gemm", &squeezellm_gemm, "Quantized GEMM for SqueezeLLM");
-
+
```
Comment (LucasWilkinson) [inline csrc/pybind.cpp:59]: nit: remove unnecessary format change

File: vllm/config.py [hunk 4]
```diff
@@ -148,9 +148,9 @@ def _verify_tokenizer_mode(self) -> None:
         self.tokenizer_mode = tokenizer_mode
 
     def _verify_sparsity(self) -> None:
-        supported_sparsity = ["sparse_w16a16"]
+        supported_sparsity = ["sparse_w16a16", "semi_structured_sparse_w16a16"]
```
Comment (mgoin) [inline vllm/config.py:151]: please rebase/merge with our main properly, it seems like you've picked up some recent changes into this diff

**Comment by rib-2** [comment]
Closing in favor of https://github.com/neuralmagic/neuralmagic-vllm/pull/26
